nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -217,22 +214,11 @@ void Optimizer::prepare(
 
 void Optimizer::shiftControlSequence()
 {
-  using namespace xt::placeholders;  // NOLINT
-  control_sequence_.vx = xt::roll(control_sequence_.vx, -1);
-  control_sequence_.wz = xt::roll(control_sequence_.wz, -1);
-
-
-  xt::view(control_sequence_.vx, -1) =
-    xt::view(control_sequence_.vx, -2);
-
-  xt::view(control_sequence_.wz, -1) =
-    xt::view(control_sequence_.wz, -2);
-
+  control_sequence_.vx = utils::rollColumns(control_sequence_.vx, 1);
```

Comment:
Does the roll also handle setting the last element to the second to last element?

Reply:
No, I'll add that part. I'll have to re-implement the rollColumns method because it is not shifting the values in place. Compiler is moving the values of the temporary matrix created inside this function, and this is causing bad memory handling, I think.

Reply:
@Ayush1285 did you complete this item?

Reply:
Yes